skeleton/nix/default.nix

```diff
@@ -0,0 +1,28 @@
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, sourcesOverride ? {}
+}:
+let
+  # use default stable nixpkgs from iohk-nix instead of our own:
+  sources = removeAttrs (import ./sources.nix) [ "nixpkgs" ] //
+    {
+      # alternatively, use iohk-nix default unstable nixpkgs:
+      #nixpkgs = iohkNix.sources.nixpkgs-unstable;
+    } //
+    sourcesOverride;
+
+  # for inclusion in pkgs:
+  nixpkgsOverlays = [
+    (_: _: { commonLib = lib // iohkNix; })
+  ];
+
+  # Import IOHK common nix lib, using our sources as override:
+  iohkNix = import sources.iohk-nix {
+    inherit system crossSystem config nixpkgsOverlays;
+    sourcesOverride = sources;
+  };
+  pkgs = iohkNix.pkgs;
```

Comment:
I still really dislike the weird proxying of the nixpkgs arguments through `iohk-nix`, but I guess that's orthogonal to this PR.

Reply:
yeah, would probably be best if iohk-nix expose an ìohk-overlays` and a `iohk-config`. and people use them explicitly.